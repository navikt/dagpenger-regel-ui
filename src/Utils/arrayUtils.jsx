export const range = length => [...Array(length).keys()];
