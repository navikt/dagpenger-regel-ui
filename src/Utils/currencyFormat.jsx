export const formatertPengesum = tall => new Intl.NumberFormat('nb-NO', { style: 'currency', currency: 'NOK' }).format(tall);
