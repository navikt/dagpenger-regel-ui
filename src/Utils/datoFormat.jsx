
export const ISO_DATE_FORMAT = 'yyyy-MM-DD';
export const DDMMYYYY_DATE_FORMAT = 'dd.MM.yyyy';
export const DDMMYY_DATE_FORMAT = 'dd.MM.YY';
export const HHMM_TIME_FORMAT = 'HH:mm';
export const YYYY_MM_FORMAT = 'yyyy-MM';
export const MMMM_YYYY_FORMAT = 'MMMM yyyy';
export const DDMMYYYYHHMM_FORMAT = 'dd.MM.yyyy HH:mm';

export const ACCEPTED_DATE_INPUT_FORMATS = [DDMMYYYY_DATE_FORMAT, 'DDMMYYYY', 'DD-MM-YYYY', 'DD/MM/yyyy'];
