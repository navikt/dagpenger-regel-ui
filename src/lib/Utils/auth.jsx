import { makeSession } from "@navikt/dp-auth";
import { azure } from "@navikt/dp-auth/identity-providers";
import { azure as azureOBO } from "@navikt/dp-auth/obo-providers";

export let getAzureSession;
getAzureSession = makeSession({
  identityProvider: azure,
  oboProvider: azureOBO,
});

export async function getInntektOboToken(session) {
  const oboToken = await session.apiToken(process.env.INNTEKT_API_AUDIENCE);
  return oboToken;
}
