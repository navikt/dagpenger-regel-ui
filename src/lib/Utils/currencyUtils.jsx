export const formatertPengesum = (tall) =>
  new Intl.NumberFormat("nb-NO", {
    style: "currency",
    currency: "NOK",
    currencyDisplay: "code", // symbol
    localeMatcher: "lookup",
  }).format(tall);

export const formatertPengesum2 = (value) => {
  const formattedValue = Number(value)
    .toLocaleString("nb-NO")
    .replace(/,|\s/g, " ");
  return `${formattedValue} kr`;
};
