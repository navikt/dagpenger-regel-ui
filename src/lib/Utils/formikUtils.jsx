import React from "react";

export function DisplayFormikState(props) {
  return (
    <div style={{ margin: "1rem 0" }}>
      <pre
        style={{
          background: "#f6f8fa",
          fontSize: "0.85rem",
          padding: "0.5rem",
          fontFamily: "monospace",
        }}
      >
        <strong>props</strong> = {JSON.stringify(props, null, 2)}
      </pre>
    </div>
  );
}

export default DisplayFormikState;
