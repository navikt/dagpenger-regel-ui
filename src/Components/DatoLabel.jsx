import React from 'react';
import PropTypes from 'prop-types';
import { formatDato } from '../Utils/dateUtils';

const DatoLabel = ({ dato, datoFormat }) => (
  <>{formatDato(new Date(dato), datoFormat)}</>

);

DatoLabel.propTypes = {
  dato: PropTypes.string.isRequired,
  datoFormat: PropTypes.string.isRequired,
};

export default DatoLabel;
