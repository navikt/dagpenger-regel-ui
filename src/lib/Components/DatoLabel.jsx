import React from "react";
import PropTypes from "prop-types";
import { formatDato } from "../Utils/datoUtils";

function DatoLabel({ dato, datoFormat }) {
  return <>{formatDato(new Date(dato), datoFormat)}</>;
}

DatoLabel.propTypes = {
  dato: PropTypes.string.isRequired,
  datoFormat: PropTypes.string.isRequired,
};

export default DatoLabel;
