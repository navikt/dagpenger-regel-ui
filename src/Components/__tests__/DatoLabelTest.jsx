import React from 'react';
import DatoLabel from '../DatoLabel';
import renderer from 'react-test-renderer';
import {MMMM_YYYY_FORMAT} from "../../Utils/datoFormat";


test('Dato label with MMMM_YYYY_FORMAT', () => {
  const component = renderer.create(
    <DatoLabel dato={'2019-05'} datoFormat={MMMM_YYYY_FORMAT}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
