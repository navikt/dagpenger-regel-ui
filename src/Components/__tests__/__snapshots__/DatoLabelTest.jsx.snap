// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dato label with MMMM_YYYY_FORMAT 1`] = `
<p
  className="typo-element"
>
  mai 2019
</p>
`;
